src/System.Text.CaseFolding/System/Text/CaseFolding/SCF.Marvin.cs

```diff
@@ -0,0 +1,136 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+
+using nuint = System.UInt64;
+
+namespace System.Text.CaseFolding
+{
+    public static partial class SCFMarvin
+    {
+        /// <summary>
+        /// Compute a Marvin hash and collapse it into a 32-bit hash.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int ComputeHash32(ReadOnlySpan<byte> data, ulong seed) => ComputeHash32(ref MemoryMarshal.GetReference(data), data.Length, (uint)seed, (uint)(seed >> 32));
+
+        /// <summary>
+        /// Compute a Marvin hash and collapse it into a 32-bit hash.
+        /// </summary>
+        public static int ComputeHash32(ref byte data, int count, uint p0, uint p1)
```

Comment:
>ComputeHash32 [](start = 26, length = 13)

from where you got this code? I think there should be away to get the Marvin hashing from the core without duplicating such code.

Reply:
The code come from CoreCLR and it is _not public_. We could see that code is just duplicated internally in CoreFX/CoreCLR https://source.dot.net/#q=Marvin

Using the code here helps us to measure performance that we could get in CoreFX/CoreCLR.

Reply:
Seems String.GetHashCode is that we need. Will try.